lib/hal_client/representation.rb

```diff
@@ -177,6 +177,18 @@ def related(link_rel, options = {}, &default_proc)
       RepresentationSet.new (Array(embedded) + Array(linked))
     end
 
+    def all_links
+      result = Set.new
+
+      embedded_entries = flatten_section(raw.fetch("_embedded", {}))
+      embedded_entries.map { |entry| result.add?(link_from_embedded_entry(entry)) }
```

Comment:
The new enumerable from `embedded_entries.map` is never used.

`result.merge(embedded_entries.map { |entry| link_from_entry(entry) })` should accomplish the goal without creating unused objects. Same for the next couple of lines.